lib/spack/spack/build_environment.py

```diff
@@ -250,25 +251,25 @@ def set_build_environment_variables(pkg, env, dirty=False):
     # Gather information about various types of dependencies
     build_deps      = pkg.spec.dependencies(deptype='build')
     link_deps       = pkg.spec.traverse(root=False, deptype=('link'))
-    build_link_deps = pkg.spec.traverse(root=False, deptype=('build', 'link'))
+    build_link_deps = set(itertools.chain(build_deps, link_deps))
     rpath_deps      = get_rpath_deps(pkg)
 
-    build_prefixes      = [dep.prefix for dep in build_deps]
+    build_run_prefixes  = [dep.prefix for dep in build_deps]
```

Comment:
Why are you calling this `build_run`? It only includes dependencies of type `build`.